onnxruntime/core/mlas/lib/intrinsics/avx512/quantize_avx512f.cpp

```diff
@@ -0,0 +1,170 @@
+    auto ZeroPointVector = _mm512_set1_epi32(ZeroPoint);
+
+    while (N >= 64) {
+
+        auto FloatVector0 = _mm512_loadu_ps(Input);
+        auto FloatVector1 = _mm512_loadu_ps(Input + 16);
+        auto FloatVector2 = _mm512_loadu_ps(Input + 32);
+        auto FloatVector3 = _mm512_loadu_ps(Input + 48);
+
+        FloatVector0 = _mm512_div_ps(FloatVector0, ScaleVector);
+        FloatVector1 = _mm512_div_ps(FloatVector1, ScaleVector);
+        FloatVector2 = _mm512_div_ps(FloatVector2, ScaleVector);
+        FloatVector3 = _mm512_div_ps(FloatVector3, ScaleVector);
+
+        FloatVector0 = _mm512_max_ps(FloatVector0, MinimumValueVector);
+        FloatVector1 = _mm512_max_ps(FloatVector1, MinimumValueVector);
+        FloatVector2 = _mm512_max_ps(FloatVector2, MinimumValueVector);
+        FloatVector3 = _mm512_max_ps(FloatVector3, MinimumValueVector);
+
+        FloatVector0 = _mm512_min_ps(FloatVector0, MaximumValueVector);
+        FloatVector1 = _mm512_min_ps(FloatVector1, MaximumValueVector);
+        FloatVector2 = _mm512_min_ps(FloatVector2, MaximumValueVector);
+        FloatVector3 = _mm512_min_ps(FloatVector3, MaximumValueVector);
+
+        auto IntegerVector0 = _mm512_cvtps_epi32(FloatVector0);
+        auto IntegerVector1 = _mm512_cvtps_epi32(FloatVector1);
+        auto IntegerVector2 = _mm512_cvtps_epi32(FloatVector2);
+        auto IntegerVector3 = _mm512_cvtps_epi32(FloatVector3);
+
+        IntegerVector0 = _mm512_add_epi32(IntegerVector0, ZeroPointVector);
+        IntegerVector1 = _mm512_add_epi32(IntegerVector1, ZeroPointVector);
+        IntegerVector2 = _mm512_add_epi32(IntegerVector2, ZeroPointVector);
+        IntegerVector3 = _mm512_add_epi32(IntegerVector3, ZeroPointVector);
+
+        _mm512_mask_cvtepi32_storeu_epi8((void*)Output, _MM_K0_REG16, IntegerVector0);
+        _mm512_mask_cvtepi32_storeu_epi8((void*)(Output + 16), _MM_K0_REG16, IntegerVector1);
+        _mm512_mask_cvtepi32_storeu_epi8((void*)(Output + 32), _MM_K0_REG16, IntegerVector2);
+        _mm512_mask_cvtepi32_storeu_epi8((void*)(Output + 48), _MM_K0_REG16, IntegerVector3);
+
+        Input += 64;
+        Output += 64;
+        N -= 64;
+    }
+
+    while (N >= 16) {
+        auto FloatVector = _mm512_loadu_ps(Input);
+        FloatVector = _mm512_div_ps(FloatVector, ScaleVector);
+        FloatVector = _mm512_max_ps(FloatVector, MinimumValueVector);
+        FloatVector = _mm512_min_ps(FloatVector, MaximumValueVector);
+
+        auto IntegerVector = _mm512_cvtps_epi32(FloatVector);
+        IntegerVector = _mm512_add_epi32(IntegerVector, ZeroPointVector);
+
+        _mm512_mask_cvtepi32_storeu_epi8((void*)Output, _MM_K0_REG16, IntegerVector);
+
+        Input += 16;
+        Output += 16;
+        N -= 16;
+    }
+
+    if (N > 0) {
+        __mmask16 mask = uint16_t((uint32_t(1) << N) - uint32_t(1));
+        auto FloatVector = _mm512_mask_loadu_ps(_mm512_set1_ps(0.0f), mask, Input);
```

Comment:
use _mm512_mask_loadu_ps instead?

Reply:
it is using _mm512_mask_loadu_ps. Do you mean somewhere to use it else?